skbio/tree/_tree.py

```diff
@@ -2430,10 +2432,14 @@ def update_result():
             starts, stops = [], []  # to calc ._start and ._stop for curr node
             for child in node.children:
                 if child.length is None:
-                    raise NoLengthError("Node with name '%s' doesn't have a "
-                                        "length." % child.name)
+                    warnings.warn("Node with name '%s' doesn't have a "
```

Comment:
It'd also be helpful to include the name of the class and method that the warning applies to (`TreeNode.tip_tip_distances`) because it may not be obvious to users where this warning is being raised from.